### DISCUSSION TITLE
New backend for TiddlyWiki5 based on Firebase

### DISCUSSION THREAD

**Discussion by neumark** [TiddlyWiki/TiddlyWiki5#5406] [General]
Hi All!

I've been using TiddlyWiki5 with Google Firebase for over a year now. It's stable enough to use on a daily basis, so I've open sourced it here: https://github.com/neumark/tw5-firebase

There's a detailed walkthrough of how to create your own instance with lots of screenshots. The entire process takes less than 30 minutes: https://neumark.github.io/tw5-firebase/

Please consider this version a proof of concept rather than a polished product! My focus was on multi-device or small-team collaboration:
- Basic authorization (bags have an access policy determining who can read / write them).
- Multiple wikis can be hosted under a single firebase account.
- It uses Firebase's built-in social auth to log in users.
- Tiddlers are written individually with locking, preventing users overwriting each others' updates.

Please let me know what you think.
Ping me at neumark [dot] peter [at] gmail [dot] com if you want access to a test wiki running on firebase or try it out under your own account for free.

All the best,
Peter

PS: sorry for the cross-post from the google group, I noticed to move to GH discussions after sending my message...

**Comment by Jermolene** [comment]
Hi @neumark I'm very pleased to see that. I've often wondered about using Firebase and/or AirTable as a back-end, and it's fascinating to see it all laid out.

May I suggest that you add a resource tiddler to `https://github.com/Jermolene/TiddlyWiki5/tree/tiddlywiki-com/editions/tw5.com/tiddlers/community/resources` (note the tiddlywiki-com branch) along the lines of [this one](https://tiddlywiki.com/#%22TiddlyWiki%20Posts%22%20by%20Jeffrey%20Kishner) to advertise it? You could pretty much copy and paste the middle part of the OP.

**Reply by neumark** [reply]
Thanks for the kind invitation, I've submitted a pull request: https://github.com/Jermolene/TiddlyWiki5/pull/5408

**Comment by makiaea** [comment]
just wanted to say thanks so much for this peter, i hope to try it out soon!

warmest wishes, maki

**Reply by neumark** [reply]
Hi @makiaea ! Good luck with it, let me know if you get stuck or have any questions!
I can provide you with a test account if you just want to try it out.
Best,
Peter
